pkg/sds/auth/universal/auth_test.go

```diff
@@ -49,90 +42,153 @@ var _ = Describe("Authentication flow", func() {
+								"kuma.io/service":  "web-api",
+								"kuma.io/protocol": "http",
 							},
 						},
 					},
 				},
 			},
 		}
-		err := resStore.Create(context.Background(), &dpRes, store.CreateBy(id.ToResourceKey()))
+		err := resStore.Create(context.Background(), &dpRes, store.CreateByKey("dp-1", "default"))
 		Expect(err).ToNot(HaveOccurred())
-
-		// when
-		credential, err := issuer.Generate(id)
-
-		// then
-		Expect(err).ToNot(HaveOccurred())
-
-		// when
-		authIdentity, err := authenticator.Authenticate(context.Background(), id, credential)
-
-		// then
-		Expect(err).ToNot(HaveOccurred())
-		Expect(authIdentity.Services[0]).To(Equal("web"))
-		Expect(authIdentity.Mesh).To(Equal(id.Mesh))
 	})
 
-	It("should throw an error on invalid token", func() {
-		// when
-		id := xds.ProxyId{
+	DescribeTable("should correctly authenticate dataplane",
+		func(id builtin_issuer.DataplaneIdentity) {
+			// when
+			credential, err := issuer.Generate(id)
+
+			// then
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			authIdentity, err := authenticator.Authenticate(context.Background(), xds.ProxyId{
+				Mesh: "default",
+				Name: "dp-1",
+			}, credential)
+
+			// then
+			Expect(err).ToNot(HaveOccurred())
+			Expect(authIdentity.Services[0]).To(Equal("web"))
+			Expect(authIdentity.Mesh).To(Equal("default"))
+		},
+		Entry("should auth with token bound to nothing", builtin_issuer.DataplaneIdentity{
+			Name: "",
+			Mesh: "",
+			Tags: nil,
+		}),
+		Entry("should auth with token bound to mesh", builtin_issuer.DataplaneIdentity{
 			Mesh: "default",
-			Name: "dp1",
-		}
-		_, err := authenticator.Authenticate(context.Background(), id, "this-is-not-valid-jwt-token")
-
-		// then
-		Expect(err).To(MatchError("could not parse token: token contains an invalid number of segments"))
-	})
-
-	It("should throw an error on token with different name", func() {
-		// when
-		generateId := xds.ProxyId{
+		}),
+		Entry("should auth with token bound to mesh and name", builtin_issuer.DataplaneIdentity{
+			Name: "dp-1",
 			Mesh: "default",
-			Name: "different-name-than-dp1",
-		}
-		token, err := issuer.Generate(generateId)
-
-		// then
-		Expect(err).ToNot(HaveOccurred())
-
-		// when
-		authId := xds.ProxyId{
+		}),
+		Entry("should auth with token bound to mesh and tags", builtin_issuer.DataplaneIdentity{
 			Mesh: "default",
-			Name: "dp1",
-		}
-		_, err = authenticator.Authenticate(context.Background(), authId, token)
-
-		// then
-		Expect(err).To(MatchError("proxy name from requestor: dp1 is different than in token: different-name-than-dp1"))
-	})
-
-	It("should throw an error on token with different mesh", func() {
-		// when
-		generateId := xds.ProxyId{
-			Mesh: "different-mesh-than-default",
-			Name: "dp1",
-		}
-		token, err := issuer.Generate(generateId)
-
-		// then
-		Expect(err).ToNot(HaveOccurred())
+			Tags: map[string]map[string]bool{
+				"kuma.io/service": {
+					"web":     true,
+					"web-api": true,
```

Comment:
Interesting, can we do these? How to I express this in curl?

Reply:
```
{
  "tags": {
    "kuma.io/service": ["web", "web-api"]
  }
}
```